src/ngraph/validation_util.cpp

```diff
@@ -651,221 +651,95 @@ PartialShape ngraph::infer_slice_shape(const Node* node,
-            {
-                begin_dms[i] = Dimension::dynamic();
-            }
-            else if (int64_t(stride_dms[i]) > 0)
-            {
-                begin_dms[i] = 0;
-            }
-            else
-            {
-                begin_dms[i] = -1;
-            }
-        }
-        else if (stride_dms[i].is_dynamic())
-        {
-            begin_dms[i] = Dimension::dynamic();
-        }
-        else if (int64_t(stride_dms[i]) > 0)
-        {
-            begin_dms[i] = 0;
-        }
-        else
-        {
-            begin_dms[i] = -1;
-        }
-
-        bj++;
-
-        if (begin_dms[i].is_static() && int64_t(begin_dms[i]) < 0)
-        {
-            begin_dms[i] = input_shape[j] + begin_dms[i];
-        }
-        // Clipping 'begin'
-        if (begin_dms[i].is_static())
-        {
-            if (int64_t(begin_dms[i]) < 0)
-            {
-                begin_dms[i] = 0;
-            }
-            else if (input_shape[j].is_dynamic())
-            {
-                begin_dms[i] = Dimension::dynamic();
-            }
-            else if (int64_t(begin_dms[i]) >= int64_t(input_shape[j]))
+    size_t input_shape_idx = 0;
+    for (size_t axis = 0; axis < begin.size(); ++axis) {
+        if (ellipsis_mask.count(axis)) { // add all dimensions hidden under the ellipsis mask if ellipsis mask is set
+            // only one bit in ellipsis mask is allowed
+            int num_new_axis_after_ellipses = 0;
+            int num_input_axis_before_ellipses = 0;
+            for (size_t i = 0; i < axis; ++i)
             {
-                begin_dms[i] = input_shape[j] - 1;
+                if (!new_axis_mask.count(i))
+                    num_input_axis_before_ellipses++;
             }
-        }
-
-        // Use upper_bounds if mask is not set
-        if (ub_mask.find(j) == ub_mask.end())
-        {
-            Dimension end_dms_tmp;
-
-            if (ub.size() <= ej)
-            {
-                end_dms_tmp = end_dms[i];
-            }
-            else if (stride_dms[i].is_dynamic())
-            {
-                end_dms_tmp = Dimension::dynamic();
-            }
-            else if (int64_t(stride_dms[i]) > 0)
-            {
-                end_dms_tmp = ub[ej] - 1;
-            }
-            else
-            {
-                end_dms_tmp = ub[ej] + 1;
+            for (size_t i = axis + 1; i < begin.size(); ++i) {
+                if (new_axis_mask.count(i))
+                    num_new_axis_after_ellipses++;
             }
 
-            if (ub.size() > ej)
-            {
-                end_dms[i] = end_dms_tmp;
-            }
-            else if (stride_dms[i].is_dynamic())
-            {
-                end_dms[i] = Dimension::dynamic();
-            }
-            else if (int64_t(stride_dms[i]) > 0)
+            unsigned long num_input_axis_after_ellipses = (begin.size() - axis - num_new_axis_after_ellipses - 1); // -1 because it's a position of ellipses
```

Comment:
```suggestion
            int64_t num_input_axis_after_ellipses = (begin.size() - axis - num_new_axis_after_ellipses - 1); // -1 because it's a position of ellipses
```